src/access/tests/TestAccessRestrictionProvider.cpp

```diff
@@ -207,49 +212,103 @@ TEST_F(TestAccessRestriction, MetaTest)
     entry.restrictions.push_back({ .restrictionType = AccessRestrictionProvider::Type::kAttributeAccessForbidden });
 
-    // must not restrict endpoint 0
-    entry.endpointNumber = 0;
+    entry.clusterId = kDescriptorCluster;
     entries.push_back(entry);
     EXPECT_EQ(accessRestrictionProvider.SetEntries(1, entries), CHIP_ERROR_INVALID_ARGUMENT);
 
-    // must not restrict network commissioning cluster
     entries.clear();
-    entry.endpointNumber = 1;
-    entry.clusterId      = kNetworkCommissioningCluster;
+    entry.clusterId = kNetworkCommissioningCluster;
     entries.push_back(entry);
     EXPECT_EQ(accessRestrictionProvider.SetEntries(1, entries), CHIP_ERROR_INVALID_ARGUMENT);
 
-    // must not restrict descriptor cluster
     entries.clear();
-    entry.clusterId = kDescriptorCluster;
+    entry.clusterId = kWiFiNetworkManagementCluster;
+    entries.push_back(entry);
+    EXPECT_EQ(accessRestrictionProvider.SetEntries(1, entries), CHIP_ERROR_INVALID_ARGUMENT);
+
+    entries.clear();
+    entry.clusterId = kThreadBorderRouterMgmtCluster;
+    entries.push_back(entry);
+    EXPECT_EQ(accessRestrictionProvider.SetEntries(1, entries), CHIP_ERROR_INVALID_ARGUMENT);
+
+    entries.clear();
+    entry.clusterId = kThreadNetworkDirectoryCluster;
+    entries.push_back(entry);
+    EXPECT_EQ(accessRestrictionProvider.SetEntries(1, entries), CHIP_ERROR_INVALID_ARGUMENT);
+
+    // also test a cluster on endpoint 0 that isnt in the special allowed list
+    entries.clear();
+    entry.clusterId = kOnOffCluster;
+    entries.push_back(entry);
+    EXPECT_EQ(accessRestrictionProvider.SetEntries(1, entries), CHIP_ERROR_INVALID_ARGUMENT);
+}
+
+// ensure no failure adding restrictions on endpoint 1 for allowed clusters only:
+// wifi network management, thread border router, thread network directory
+TEST_F(TestAccessRestriction, ValidRestrictionsOnEndpointOneTest)
+{
+    std::vector<AccessRestrictionProvider::Entry> entries;
+    AccessRestrictionProvider::Entry entry;
+    entry.endpointNumber = 1;
+    entry.fabricIndex    = 1;
+    entry.restrictions.push_back({ .restrictionType = AccessRestrictionProvider::Type::kAttributeAccessForbidden });
+
+    entry.clusterId = kWiFiNetworkManagementCluster;
+    EXPECT_EQ(accessRestrictionProvider.SetEntries(1, entries), CHIP_NO_ERROR);
+
+    entries.clear();
+    entry.clusterId = kThreadBorderRouterMgmtCluster;
+    entries.push_back(entry);
+    EXPECT_EQ(accessRestrictionProvider.SetEntries(1, entries), CHIP_NO_ERROR);
+
+    entries.clear();
+    entry.clusterId = kThreadNetworkDirectoryCluster;
+    entries.push_back(entry);
+    EXPECT_EQ(accessRestrictionProvider.SetEntries(1, entries), CHIP_NO_ERROR);
+
+    // also test a cluster on endpoint 0 that isnt in the special allowed list
+    entries.clear();
+    entry.clusterId = kOnOffCluster;
+    entries.push_back(entry);
+    EXPECT_EQ(accessRestrictionProvider.SetEntries(1, entries), CHIP_ERROR_INVALID_ARGUMENT);
+}
```

Comment:
This doesn't actually set the entry on endpoint 0. This should not fail...

Reply:
done